### ISSUE TITLE
Dragon unlock checks

### ISSUE THREAD

**Issue by GeneralYouri** [eternaldensity/Sandcastle-Builder#567]
The function Molpy.CheckDragon() is responsible for checking whether AD's power is high enough to unlock a new Dragon boost. However, this check is only performed when destroying duplicates. However, all of these boosts have additional requirements besides the AD level. This leads boosts to sometimes not unlock until way later in the game, when the required AD level is achieved before the other requirements.

In my case, once I got to infinite sand/castles and started working with the TF, I wasn't really working on my duplicates anymore at all. I had >1k power, enough for Crystal Dragon, but was far away from the other requirement: have Logic (1200/Panther Rush uses). But now that I easily meet this requirement, the boost still didn't unlock, and it wasn't until I checked the code that I found this.

Long story short - unlock checks for the dragon boosts (and potentially other boosts with multiple requriements aswell) should be performed when any of the requirements is updated, not just one.
